core-web/libs/sdk/angular/next/services/dotcms-editable-page.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Observable, of, Subject } from 'rxjs';
+
+import { Injectable } from '@angular/core';
+
+import { finalize } from 'rxjs/operators';
+
+import { updateNavigation } from '@dotcms/client';
+import { UVEEventType, DotCMSEditablePage } from '@dotcms/types';
+import { createUVESubscription, getUVEState, initUVE } from '@dotcms/uve';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class DotCMSEditablePageService {
+    /**
+     * Subject that emits the current editable page asset or null.
+     * Used internally to track changes to the page data.
+     *
+     * @private
+     * @type {Subject<DotCMSEditablePage | null>}
+     */
+    #pageAssetSubject = new Subject<DotCMSEditablePage | null>();
+
+    /**
+     * Observable stream of the page asset changes.
+     * Exposes the pageAssetSubject as an Observable for subscribers.
+     *
+     * @private
+     * @type {Observable<DotCMSEditablePage | null>}
+     */
+    #pageAsset$ = this.#pageAssetSubject.asObservable();
+
+    /**
+     * Listens for changes to an editable page and returns an Observable that emits the updated page data.
+     * This method initializes the UVE (Universal Visual Editor) and sets up subscriptions to track content changes.
+     *
+     * @example
+     * ```ts
+     * // Import the service
+     * import { DotCMSEditablePageService } from '@dotcms/angular';
+     *
+     * // Inject the service
+     * constructor(private editablePageService: DotCMSEditablePageService) {}
+     *
+     * // Get the page data from your API call
+     * const page = await client.page.get('/');
+     *
+     * // Listen for changes
+     * const subscription = this.editablePageService.listen(page).subscribe(updatedPage => {
+     *   if (updatedPage) {
+     *     // Handle updated page data
+     *     console.log('Page updated:', updatedPage);
+     *   }
+     * });
+     *
+     * // When done listening, unsubscribe
+     * subscription.unsubscribe();
+     * ```
+     *
+     * @param pageAsset Optional initial page data
+     * @returns Observable that emits the updated page data or null
+     */
+    listen(pageAsset?: DotCMSEditablePage): Observable<DotCMSEditablePage | null> {
+        if (!getUVEState()) {
+            return of(pageAsset || null);
+        }
+
+        const pageURI = pageAsset?.page?.pageURI ?? '/';
+
+        initUVE(pageAsset);
+        updateNavigation(pageURI);
```

Comment:
Is updating the Navigation here necessary? By default, the UVE already opens the page specified by the PageURI.

Reply:
We need it, because when you navigate internally, this should load again and trigger a navigation on the UVE, to keep the data on sync